### PR TITLE
[GDI32] Fix return value based on GetObject gdi32_apitest

### PR DESCRIPTION
**Pull request by zecarlos1957**
Fix returned error code value  based on GetObject gdi32_apitest

### CODE REVIEW

**Comment by zecarlos1957** [comment]
VirtualBox_WinSrv2003_Gdi32_apitest_GetObject

**Comment by zecarlos1957** [comment]
VirtualBox_ReactOS_Gdi32_apitest_GetObject

File: win32ss/gdi/gdi32/objects/gdiobj.c [hunk 1]
```diff
@@ -265,10 +265,11 @@ GetObjectW(
                 (dwType == GDI_OBJECT_TYPE_BRUSH) ||
                 (dwType == GDI_OBJECT_TYPE_COLORSPACE) ||
                 ( (dwType == GDI_OBJECT_TYPE_EXTPEN) &&
-                    ( (cbSize >= sizeof(EXTLOGPEN)) || (cbSize == 0) ) ) ||
+                    ( (cbSize >= sizeof(EXTLOGPEN)))) ||
                 ( (dwType == GDI_OBJECT_TYPE_BITMAP) && (cbSize >= sizeof(BITMAP)) ))
             {
-                SetLastError(ERROR_NOACCESS);
+                if (cbSize)
```
Comment (HBelusca) [inline win32ss/gdi/gdi32/objects/gdiobj.c:271]: So what's the error set when cbSize == 0? none (stays as the ambient one)?

Comment (zecarlos1957) [reply win32ss/gdi/gdi32/objects/gdiobj.c:271]: This is the result observed in Win2003Srv, but I suspect it should be ERROR_INVALID_PARAMETER, as the function returns 0 indicating an error condition.
Win11 has a different result (with the original code).

CapturaWin11GetObject

Comment (zecarlos1957) [reply win32ss/gdi/gdi32/objects/gdiobj.c:271]: Yes, this is the behavior of Win2003 and WinXP. Only Win11 has a different behavior, as in the present case, with a válid HBRUSH hand
CapturaWin11GetObject
ler, szSize = 0 and a válid value in the buffer address the function returns the sizeof(LOGBRUSH) == 0x0c.

**Review by DarkFire01 — APPROVED** [review]

**Review by JoachimHenze — APPROVED** [review]
Code structure looks a bit weird. I do wonder whether cbSize should be checked a bit earlier instead. But approved, until proven to be wrong/suboptimal by future tests.

The test of GetObject seems to be ok, all 344 tests do also pass the WHS bot:
https://reactos.org/testman/compare.php?ids=96254

**Comment by JoachimHenze** [comment]
0.4.15-dev-8277-gcdb4385 vs. 0.4.15-dev-8278-gc3a45b6:
VBox https://reactos.org/testman/compare.php?ids=96311,96313 a lot of unrelated? fluctuations currently on the bots
KVM https://reactos.org/testman/compare.php?ids=96304,96308 looks a bit better, but still too much noise to draw any final conclusions

**Pull request merged**